api/src/main/java/org/apache/iceberg/util/CharSequenceWrapper.java

```diff
@@ -49,21 +50,22 @@ public boolean equals(Object other) {
     if (this == other) {
       return true;
     }
-    if (other == null || getClass() != other.getClass()) {
+    if (other == null) {
       return false;
     }
 
-    CharSequenceWrapper that = (CharSequenceWrapper) other;
-    return Comparators.charSequences().compare(wrapped, that.wrapped) == 0;
+    if (other instanceof CharSequence) {
+      return Comparators.charSequences().compare(wrapped, (CharSequence) other) == 0;
+    } else if (other instanceof CharSequenceWrapper) {
+      CharSequenceWrapper that = (CharSequenceWrapper) other;
+      return Comparators.charSequences().compare(wrapped, that.wrapped) == 0;
+    } else {
+      return false;
+    }
   }
 
   @Override
   public int hashCode() {
-    int result = 177;
-    for (int i = 0; i < wrapped.length(); i += 1) {
-      char ch = wrapped.charAt(i);
-      result = 31 * result + (int) ch;
-    }
-    return result;
+    return wrapped.hashCode();
```

Comment:
Why did this change?

There is no guarantee that implementations of `CharSequence` produce the same `hashCode`, so this breaks the contract of equals/hashCode.

Reply:
Thanks for the comments. Done.